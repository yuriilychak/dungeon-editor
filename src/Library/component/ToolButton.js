import React from "react";
import {makeStyles} from '@material-ui/styles';
import IconButton from '@material-ui/core/IconButton';
import Tooltip from '@material-ui/core/Tooltip';


const useStyles = makeStyles({
    button: {
        fontSize: 20,
        padding: 0
    },
    icon: {
        fontSize: 20,
        fill: "#ffffff"
    }
});

const ToolButton = ({title, onClick, Icon}) => {
    const classes = useStyles();
    return (
        <Tooltip title={title}>
            <IconButton className={classes.button} onClick={onClick}>
                <Icon className={classes.icon} />
            </IconButton>
        </Tooltip>
    );
};

export default ToolButton;
