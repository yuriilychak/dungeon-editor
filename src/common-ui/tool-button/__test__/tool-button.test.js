import React from "react";
import { createMount } from "@material-ui/core/test-utils";
import ToolButton from "../tool-button";

describe("ToolButton test with icon",()=> {

    const Icon = () => <div>Icon</div>;
    const onClick = jest.fn();
    const owner = "test";

    it("default snapshot", () => {
        const wrapper = createMount()(
            <ToolButton
            Icon={Icon}
            owner={owner}
            onClick={onClick}
        />
        );
        expect(wrapper.html()).toMatchSnapshot();
    });

    it("with tooltip snapshot", () => {
        const wrapper = createMount()(
            <ToolButton
                Icon={Icon}
                owner={owner}
                onClick={onClick}
                title="title"
            />
        );
        expect(wrapper.html()).toMatchSnapshot();
    });

    it("test click", () => {
        const wrapper = createMount()(
            <ToolButton
                Icon={Icon}
                owner={owner}
                onClick={onClick}
            />
        );

        wrapper.find("button").simulate("click");
        expect(onClick).toHaveBeenCalled();
    });
});


