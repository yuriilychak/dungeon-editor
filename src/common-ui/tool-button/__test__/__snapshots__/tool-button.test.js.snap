// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToolButton test with icon default snapshot 1`] = `"<button class=\\"MuiButtonBase-root MuiIconButton-root\\" tabindex=\\"0\\" type=\\"button\\" aria-owns=\\"test\\"><span class=\\"MuiIconButton-label\\"><div>Icon</div></span><span class=\\"MuiTouchRipple-root\\"></span></button>"`;

exports[`ToolButton test with icon with tooltip snapshot 1`] = `"<button class=\\"MuiButtonBase-root MuiIconButton-root\\" tabindex=\\"0\\" type=\\"button\\" aria-owns=\\"test\\" aria-haspopup=\\"true\\" title=\\"title\\"><span class=\\"MuiIconButton-label\\"><div>Icon</div></span><span class=\\"MuiTouchRipple-root\\"></span></button>"`;
